### ISSUE TITLE
Updating any data repaints the whole table !!

### ISSUE THREAD

**Issue by varun-dev** [handsontable/handsontable#3147]
If I make a call to setDataAtCell or populateFromArray to change data, the whole table gets repainted. Is this expected behaviour ?

Here is a fiddle to demonstrate it. 
- I create a table with data
- I add a red background color inline to all the td elements
- then I make a call to setDataAtCell to change one value on the click of a button

http://jsfiddle.net/varun_dev/n94n8dn4/

You will notice that all the td tags have lost their inline red colour which I assume means that every cell is getting repainted ??

**Comment by thip** [comment]
I think you probably want to be using a custom renderer to set the cell background. Have a look at these:

[http://docs.handsontable.com/0.21.0/demo-custom-renderers.html](http://docs.handsontable.com/0.21.0/demo-custom-renderers.html)
[http://docs.handsontable.com/0.21.0/demo-conditional-formatting.html](http://docs.handsontable.com/0.21.0/demo-conditional-formatting.html)

**Comment by AMBudnik** [comment]
Hi @varun-dev , @thip is right. You should try to use conditional formatting.I think it is changing all your cells cause of rerendering.

**Comment by varun-dev** [comment]
Thanks for your responses. I actually didn't want to have red color as background but it was just to demonstrate that all the TDs are re-rendering when I am just updating few elements.

Here is my updated demo. I have added a custom cell renderer for all the cells. Its just adds a log and calls the default renderer. Now when I update one cell, the cell renderers for every cell are getting called as you will notice in the console. This is a hit on performance and is there any way to avoid re-processing every cell when I have just updated one cell ??

https://jsfiddle.net/varun_dev/n94n8dn4/
